### ISSUE TITLE
Ordering of scripts in build is incorrect resulting in broken paper-buttons

### ISSUE THREAD

**Issue by sorvell** [Polymer/polymer-element-catalog#82]
The build vulcanizes elements.html and each loads the following before doing any execution so read this list backwards:

1. elements.html
2. app-shell.html
3. app-cart.html
4. paper-button.html
5. paper-button-behavior.html
 defines: 
```PaperButtonBehavior = [
    Polymer.IronButtonState, // loaded below
    Polymer.IronControlState, // loaded below
    Polymer.PaperButtonBehaviorImpl // in this file
];
```
6. iron-button-state.html // defines Polymer.IronButtonState
7. iron-control-state.html // defines Polymer.IronControlState

But in the build 5 is before 6 and 7. This seems like vulcanize ordering is messed up.

**Comment by sorvell** [comment]
Console warnings:

```
[paper-icon-button::_flattenBehaviorsList]: behavior is null, check for missing or 404 import
2elements.build.js:217 [paper-button::_flattenBehaviorsList]: behavior is null, check for missing or 404 import
```

**Comment by sorvell** [comment]
If you click the cart icon and then the download tab, you'll see that the button has no shadow. This is a symptom of this problem.

**Comment by azakus** [comment]
@garlicnation this is confusing. If I run vulcanize on `paper-button.html`, the order is correct. Maybe something is importing `iron-control-state` in a weird order?

**Comment by azakus** [comment]
Hmm, I have a hunch that hydrolysis is marking duplicate imports incorrectly by picking the "deepest" copy as the winner, but vulcanize uses replaceWith in head inside-out, which can lead to dependencies being mismatched.

I constructed a slim version of the `app-shell` import tree that breaks. If I log the ordering of `flatten` in vulcanize, I see that paper-behaviors.html is loaded way before the paper-radio-behaviors loads the iron-control-state behavior.

I made a gist with my setup: https://gist.github.com/azakus/1cd70027ab0f72c1ee71

**Comment by azakus** [comment]
Looking at the HTML Imports spec, It looks like dependencies should be resolved in a breadth-first order. `hydrolysis.Analyzer._metadataTree` looks to be doing a depth-first traversal: http://w3c.github.io/webcomponents/spec/imports/#import-dependent

**Comment by azakus** [comment]
No, I think I proved to myself I'm wrong about that. I still have a hunch that something is wrong in the traversal of imports, but it may be in vulcanize instead.

**Comment by azakus** [comment]
Filed https://github.com/Polymer/hydrolysis/issues/116

**Comment by azakus** [comment]
Fixed with `hydrolysis 1.12.0` and `vulcanize 1.8.1`
